### ISSUE TITLE
OOM when calling pretrained models on sequences

### ISSUE THREAD

**Issue by wward97** [songlab-cal/tape#88]
Hi! I have been using 2 pretrained models, the transformer and UNIREP, to get representations for a supervised model. The eventual goal will be to write a training loop to fine tune the models to my specific data, however I am running into memory issues when just passing the models a couple sequences. I saw the update about recommending switching APIs for training and I will follow that however I was wondering if you have any thoughts as to why I can barely get representations from the pretrained models. I have 16GB of GPU and can expand to more with AWScloud, I think I can pass maybe 4 sequences to UNIREP before it gets too much for it, and 40 for the transformer.  I am using:
```
model = ProteinBertModel.from_pretrained('bert-base')
tokenizer = TAPETokenizer(vocab='iupac') 
```
and then turning my tokenized sequences into pytorch tensors, moving them to GPU and moving the model to GPU and then running the model. Is this a normal limitation?

**Comment by rmrao** [comment]
Are you using `torch.no_grad`?

**Comment by wward97** [comment]
```
`token_ids = token_ids.to(device)[:10]

print(np.shape(token_ids), token_ids.type())

rep = None

print('Entering Loop')

print('-------------------')

for sequence in batch(token_ids,5):

    print(sequence.requires_grad)

    print("Size of input sequence: ", np.shape(sequence), "Type of sequence: ", sequence.type())

    output = model(sequence)

    print(output[0].requires_grad)`

```

So i check if sequence requires grad and I get false, I check the data type and size of the sequence before I put it into the model and these are the outputs:

torch.Size([10, 240]) torch.LongTensor
Entering Loop
-------------------
False
Size of input sequence:  torch.Size([5, 240]) Type of sequence:  torch.LongTensor

and then it usually crashes in the model section, where the model code is in the original code. The output when i tested this with the transformer indeed had the requires_grad tag as true and so i removed it so i could delete the values after I save it so I can run the model. However as mentioned the batch size is quite small.

**Comment by rmrao** [comment]
Hmm. I'm not totally sure, I think you should be able to run that sequence?

I'd try adding `torch.set_grad_enabled(False)` or putting everything inside a `with torch.no_grad()` context. It's possible it's storing gradients for the model parameters as well.

If that doesn't work, that may be the limit. 40 sequences for the transformer sounds about right, 5 sequences for UniRep seems low though.

**Comment by rmrao** [comment]
Closing b/c no response. Please reopen w/ more info if it still doesn't work.
